agent/host.go

```diff
@@ -52,6 +54,59 @@ type CPUInfo struct {
 	Microcode  string `json:"Microcode,omitempty"`
 }
 
+// Extract class and ID from the children node
+func extractClassAndID(children map[string]interface{}) (string, string) {
+	var foundID string
+	var foundClass string
+	for objectName, objectValue := range children {
+		switch objectValue.(type) {
```

Comment:
More idiomatic writting:

```
switch objectValue := objectValue.(type) {
case string:
        switch objectName {
        "class": 
                  ...
        "id":
                  ....
         }
          ...
default:
          // log error
}
```

Reply:
Yes, the switch is a remaining of the first implementation but have the interest of being more flexible if I have to reprocess some other structures. So yes it's a bit overkill for the current code but useful for testing/adding new features or quirks.